### ISSUE TITLE
CDC heap memory is always rise up (OOM) when upstream has no I/O

### ISSUE THREAD

**Issue by Tammyxia** [pingcap/ticdc#1904]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.

- Create two changefeed with different downstream, but has the same source database.
- "sysbench oltp_common prepare  ... --threads=512 --tables=1000 --table-size=0" to upstream
- Check CDC status.

2. What did you expect to see?

- No OOM, sync task works as expected.

3. What did you see instead?

- heap memory and goroutine is always rise up, even when sysbench is stopped, CDC server VM memory is 64GB, but heap memory can be 100GB+, oom risk.
 
![image](https://user-images.githubusercontent.com/78345569/120271603-37d27600-c2de-11eb-9367-0c1c67cc9a84.png)

- cdc.log
 
![image](https://user-images.githubusercontent.com/78345569/120272047-fabab380-c2de-11eb-84d3-d100a01a1835.png)

- kv log
 
![image](https://user-images.githubusercontent.com/78345569/120272721-1b373d80-c2e0-11eb-8248-a28025ed4765.png)

4. Versions of the cluster

    - Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

        ```
        Release Version: v5.1.0-alpha-13-gd5e0ed0aa-dirty
Edition: Community
Git Commit Hash: d5e0ed0aaed72d2f2dfe24e9deec31cb6cb5fdf0
Git Branch: master
UTC Build Time: 2021-05-24 14:39:20
GoVersion: go1.13
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false |

        ```

    - TiCDC version (execute `cdc version`):

        ```
        [release-version=v5.0.0-master-dirty] [git-hash=ca64c784185c8df774cdd19444e3d4dac1b5f6c5] [git-branch=master] [utc-build-time="2021-05-31 05:29:05"] [go-version="go version go1.16.4 linux/amd64"] [failpoint-build=false]
        ```

**Comment by Tammyxia** [comment]
Listening  1.49k table
![image](https://user-images.githubusercontent.com/78345569/120274121-fba11480-c2e1-11eb-8b27-fd20d3b20ee0.png)

**Comment by amyangfei** [comment]
Bug only exists in master, in fact it is not an OOM issue, it is caused by some goroutines are not recycled

**Comment by amyangfei** [comment]
Fixed in #1903
